python/heterocl/schedule.py

```diff
@@ -31,14 +32,29 @@ class Schedule(object):
     """
 
     stage_ops = []
+    stage_names = set()
+    mod_calls = dict()
     last_stages = OrderedSet([])
     _ids = count(0)
 
-    def __init__(self, sch, inputs, name=""):
+    def __init__(self, sch, inputs, outputs, name=""):
         self.id = next(self._ids)
         self.sch = sch
-        self.inputs = inputs
-        self.placement = dict()
+        self.inputs = inputs + outputs
+        self.outputs = outputs
+
+        # tensor on hold for chained primitives
+        self.hold_tensor = None
+        self.hold_source_stage = None
```

Comment:
Can you explain what is "tensor on hold"?

Reply:
It is used to support the chained .to(). For example `s.to(A, p.xcel).to(XXX)`, the second `.to` does not have the target tensor specified, so we need to have the target tensor saved somewhere.